src/IncrementalChecker.ts

```diff
@@ -238,7 +238,10 @@ class IncrementalChecker {
       try {
         this.linter.lint(fileName, undefined, this.linterConfig);
       } catch (e) {
-        if (fs.existsSync(fileName)) {
+        if (
+          fs.existsSync(fileName) &&
+          !e.message.trim().startsWith("Invalid source file")
```

Comment:
Are we sure that linter always throws an "Error" like object with a message? I think it would be safer to add the additional check:
```
fs.existsSync(fileName) &&
!(e.message && e.message.trim().startsWith("Invalid source file"))
```
Maybe there is a better way to check if given error is about invalid source file - maybe we could add `instanceof` check instead of message matching? (just asking - not sure how it's implemented on tslint side).
The last thing - please add a comment here - it's not obvious that there could be a filesystem lag :)

Reply:
I'll have to boot up my machine at work Monday to look again at it again. If I remember correctly it is just throwing a string as error, but I'll give it another look. Thanks for quick response boss man I'll get it all figured out!